File: jserver-core/src/test/java/com/github/gr3gdev/jserver/http/impl/RequestImplTest.java
package com.github.gr3gdev.jserver.http.impl;

import org.junit.Test;

import java.io.IOException;
import java.io.InputStream;
import java.util.Optional;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class RequestImplTest {

    private void validate(Optional<String> key, String value) {
        assertTrue(key.isPresent());
        key.ifPresent(it -> assertEquals(value, it));
    }

    @Test
    public void testGetRequest() throws IOException {
        try (InputStream it = RequestImplTest.class.getResourceAsStream("/http/get.txt")) {
            final RequestImpl req = new RequestImpl("", it);
            assertEquals("/test", req.path());
            assertEquals(4, req.headersNames().size());
            validate(req.headers("User-Agent"), "Mozilla/4.0");
            validate(req.headers("Host"), "www.example.com");
            validate(req.headers("Accept-Language"), "en, fr");
            validate(req.headers("Connection"), "Keep-Alive");
            assertEquals(2, req.paramsNames().size());
            validate(req.params("param1"), "1");
            validate(req.params("param2"), "why");
        }
    }

    @Test
    public void testPostRequest() throws IOException {
        try (InputStream it = RequestImplTest.class.getResourceAsStream("/http/post.txt")) {
            final RequestImpl req = new RequestImpl("", it);
            assertEquals("/login", req.path());
            assertEquals(7, req.headersNames().size());
            validate(req.headers("User-Agent"), "Mozilla/4.0");
            validate(req.headers("Host"), "www.example.com");
            validate(req.headers("Accept-Language"), "en, fr");
            validate(req.headers("Connection"), "Keep-Alive");
            validate(req.headers("Content-Type"), "application/x-www-form-urlencoded");
            validate(req.headers("Content-Length"), "length");
            validate(req.headers("Accept-Encoding"), "gzip, deflate");
            assertEquals(2, req.paramsNames().size());
            validate(req.params("username"), "person");
            validate(req.params("password"), "secret");
        }
    }

}

File: jserver-core/src/test/kotlin/com/github/gr3gdev/jserver/http/impl/RequestImplTest.kt
package com.github.gr3gdev.jserver.http.impl

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test
import java.util.*

class RequestImplTest {

    private fun validate(key: Optional<String>, value: String) {
        assertTrue(key.isPresent)
        key.ifPresent {
            assertEquals(value, it)
        }
    }

    @Test
    fun `Test get request`() {
        javaClass.getResourceAsStream("/http/get.txt").use {
            val req = RequestImpl("", it)
            assertEquals("/test", req.path())
            assertEquals(4, req.headersNames().size)
            validate(req.headers("User-Agent"), "Mozilla/4.0")
            validate(req.headers("Host"), "www.example.com")
            validate(req.headers("Accept-Language"), "en, fr")
            validate(req.headers("Connection"), "Keep-Alive")
            assertEquals(2, req.paramsNames().size)
            validate(req.params("param1"), "1")
            validate(req.params("param2"), "why")
        }
    }

    @Test
    fun `Test post request`() {
        javaClass.getResourceAsStream("/http/post.txt").use {
            val req = RequestImpl("", it)
            assertEquals("/login", req.path())
            assertEquals(7, req.headersNames().size)
            validate(req.headers("User-Agent"), "Mozilla/4.0")
            validate(req.headers("Host"), "www.example.com")
            validate(req.headers("Accept-Language"), "en, fr")
            validate(req.headers("Connection"), "Keep-Alive")
            validate(req.headers("Content-Type"), "application/x-www-form-urlencoded")
            validate(req.headers("Content-Length"), "length")
            validate(req.headers("Accept-Encoding"), "gzip, deflate")
            assertEquals(2, req.paramsNames().size)
            validate(req.params("username"), "person")
            validate(req.params("password"), "secret")
        }
    }

}